### PR TITLE
remove EventGridFaultyEvent since they fixed the server side

### PR DESCRIPTION
**Pull request by watashiSHUN**

### CODE REVIEW

File: src/EventGridExtension/EventGridExtension.csproj [hunk 1]
```diff
@@ -11,13 +11,13 @@
   </PropertyGroup>
 
   <ItemGroup Condition="$(TargetFramework)=='net46'">
-      <PackageReference Include="Microsoft.Azure.WebJobs" Version="2.1.0-beta1" />
-      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions" Version="2.1.0-beta1" />
-      <Reference Include="System.Web" />
+    <PackageReference Include="Microsoft.Azure.WebJobs" Version="2.1.0-beta1-10976" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions" Version="2.1.0-beta1-10489" />
```
Comment (watashiSHUN) [inline src/EventGridExtension/EventGridExtension.csproj]: @MikeStall I updated the version, we can rebase your tests

File: test/localTests/Function.cs [hunk 2]
```diff
@@ -10,27 +11,18 @@ public class Function
...
-        {
-            foreach (var b in myBlob)
-            {
-                Console.Write(b);
-            }
+            Console.WriteLine(value);
         }
 
-        public void TestString([EventGridTrigger("eventhubcapture", Connection = "ShunTestConnectionString")] string myBlob)
+        public void TestBlobStream([EventGridTrigger] EventGridEvent value, [Blob("{data.container}/{data.blob}", FileAccess.Read, Connection = "ShunTestConnectionString")]Stream myBlob)
```
Comment (watashiSHUN) [inline test/localTests/Function.cs]: @christopheranderson, new signature

File: src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs [hunk 3]
```diff
@@ -38,49 +40,66 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
...
-            // unsupported publisher is caught in attribute constrcutor
-            return Task.FromResult<ITriggerBinding>(new EventGridTriggerBinding(context.Parameter, _extensionConfigProvider, context.Parameter.Member.Name, publisher, contract));
+
+            return Task.FromResult<ITriggerBinding>(new EventGridTriggerBinding(context.Parameter, _extensionConfigProvider, context.Parameter.Member.Name, contract));
 
         }
 
+        public Dictionary<string, Type> ExtractBindingContract(Type t)
+        {
+            if (t == typeof(EventGridEvent) || t == typeof(string))
```
Comment (MikeStall) [inline src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs]: The binding data contract here should be agnostic to the user's parameter type.

File: src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs [hunk 4]
```diff
@@ -38,49 +40,66 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
...
                 _bindingContract = contract;
             }
+            public Task<Dictionary<string, object>> ExtractBindingData(EventGridEvent e, Type t)
+            {
+                var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+                if (t == typeof(EventGridEvent))
+                {
+                    bindingData.Add("EventGridTrigger", e);
+                }
+                else if (t == typeof(string))
```
Comment (MikeStall) [inline src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs]: The binding data != parameters.  Separate concepts. The binding data is like auxiallary (language agnostic) information in addition to the parameter.

File: src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs [hunk 5]
```diff
@@ -38,49 +40,66 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
...
+                {
+                    bindingData.Add("EventGridTrigger", JsonConvert.SerializeObject(e, Formatting.Indented));
+                }
+                bindingData.Add("data", e.Data);
+
+                return Task.FromResult<Dictionary<string, object>>(bindingData);
+            }
+            public object GetArgument(Dictionary<string, object> bindingData)
```
Comment (MikeStall) [inline src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs]: so we shouldn't get the argument from the binding data..

File: src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs [hunk 6]
```diff
@@ -96,8 +115,8 @@ public Type TriggerValueType
             public async Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
             {
                 EventGridEvent triggerValue = value as EventGridEvent;
-                var bindingData = await _publisher.ExtractBindingData(triggerValue, _parameter.ParameterType);
-                IValueBinder valueBinder = new EventGridValueBinder(_parameter, _publisher.GetArgument(bindingData), _publisher.Recycles);
+                var bindingData = await ExtractBindingData(triggerValue, _parameter.ParameterType);
+                IValueBinder valueBinder = new EventGridValueBinder(_parameter, GetArgument(bindingData));
```
Comment (MikeStall) [inline src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs]: Don't have GetArgument(bindingData). Instead, just pass value directly.  If we need to coerce it to the user's parameter type, we can coerce it here.

File: test/localTests/Function.cs (hunk 2 above)
Comment (MikeStall) [inline test/localTests/Function.cs]: Perfect!

**Comment by watashiSHUN** [comment]
moved to https://github.com/Azure/azure-functions-eventing-extension/pull/9

**Pull request closed without merging**